### PR TITLE
Armando/grwth 3385 trending mints api endpoint

### PR DESCRIPTION
**Pull request by r3lays**

### CODE REVIEW

**Review by nofir — CHANGES_REQUESTED** [review]

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+const REDIS_EXPIRATION = 60 * 60 * 24; // 24 hours
+const REDIS_EXPIRATION_MINTS = 120; // Assuming an hour, adjust as needed.
+
+import { getTrendingMints } from "@/elasticsearch/indexes/activities";
+
+import { getJoiPriceObject, JoiPrice } from "@/common/joi";
+import { Sources } from "@/models/sources";
+
+const version = "v1";
+export interface MintResult {
```
[the remark this answers is not in the document]
Comment (r3lays) [reply packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:23]: Are you referring to the version or the sources? 
The "version" is used in the handler log statements & the "Sources" is used on line 328 to get the source for the floor ask.

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+        floorAsk,
+      };
+    })
+  );
+
+  return collections;
+}
+
+async function getCollectionsMetadata(collectionIds: string[]): Promise<Record<string, Metadata>> {
+  const collectionsToFetch = collectionIds.map((id: string) => `collection-cache:v1:${id}`);
```
[the remark this answers is not in the document]
Comment (r3lays) [reply packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts]: I merged the two functions together.
Have to discuss with Michael re the caching logic.

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+      c.day1_volume_change,
+      c.day7_volume_change,
+      c.day30_volume_change,
+      c.all_time_volume,
+      json_build_object(
+          'imageUrl', c.metadata ->> 'imageUrl',
+          'bannerImageUrl', c.metadata ->> 'bannerImageUrl',
+          'description', c.metadata ->> 'description'
+      ) AS metadata,
+      c.non_flagged_floor_sell_id,
```
[the remark this answers is not in the document]
Comment (r3lays) [reply packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts]: Gonna bring this up to Michael as well.

**Review by lunarmayor — DISMISSED** [review]

File: packages/indexer/src/api/routes.ts [hunk 4]
```diff
@@ -514,6 +514,12 @@ export const setupRoutes = (server: Server) => {
     options: collectionsEndpoints.getTopSellingCollectionsV2Options,
   });
 
+  server.route({
+    method: "GET",
+    path: "/collections/get-trending-mints/v1",
```
Comment (nofir) [inline packages/indexer/src/api/routes.ts:519]: should be (to follow the same pattern):
/collections/trending-mints/v1

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+  `;
+
+  const result = await redb.manyOrNone<Mint>(baseQuery);
+
+  await redis.set(cacheKey, JSON.stringify(result), "EX", REDIS_EXPIRATION_MINTS);
+
+  return result;
+}
+
+async function formatCollections(
```
Comment (nofir) [inline packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:237]: can this use the formatCollections from the trending and add the mint data? if not, its fine.

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+      throw error;
+    },
+  },
+  handler: async ({ query }: Request, h) => {
+    const { normalizeRoyalties, useNonFlaggedFloorAsk, type, period, limit } = query;
+
+    try {
+      const mintingCollections = await getMintingCollections(type);
+
+      const elasticMintData = await getTrendingMints({
```
Comment (nofir) [inline packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:143]: I would have a check here on mintingCollections.length to make sure we don't call es for nothing

**Review by nofir — CHANGES_REQUESTED** [review]

**Review by nofir — APPROVED** [review]

**Pull request merged**